src/database.ts

```diff
@@ -1386,6 +1525,90 @@ class Database extends GrpcServiceObject {
     });
     return waitForSessionStream;
   }
+
+  restore(backupPath: string): Promise<RestoreDatabaseResponse>;
+  restore(
+    backupPath: string,
+    options?: CallOptions
+  ): Promise<RestoreDatabaseResponse>;
+  restore(backupPath: string, callback: RestoreDatabaseCallback): void;
+  restore(
+    backupPath: string,
+    options: CallOptions,
+    callback: RestoreDatabaseCallback
+  ): void;
+  /**
+   * @typedef {array} RestoreDatabaseResponse
+   * @property {Database} 0 The new {@link Database}.
+   * @property {Operation} 1 An {@link Operation} object that can be used to check
+   *     the status of the request.
+   * @property {object} 2 The full API response.
+   */
+  /**
+   * @callback RestoreDatabaseCallback
+   * @param {?Error} err Request error, if any.
+   * @param {Database} database The new {@link Database}.
+   * @param {Operation} operation An {@link Operation} object that can be used to
+   *     check the status of the request.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Restore a backup into this database.
+   *
+   * When this call completes, the restore will have commenced but will not
+   * necessarily have completed.
+   *
+   * @param backupPath the path of the backup to restore.
+   * @param {object} [options] Configuration object.
+   * @returns Promise<RestoreDatabaseResponse> when resolved, contains the restore operation.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = instance.database('my-database');
+   * const backupName = 'projects/my-project/instances/my-instance/backups/my-backup';
+   * const [, restoreOperation] = await database.restore(backupName);
+   * // Wait for restore to complete
+   * await restoreOperation.promise();
+   */
+  restore(
+    backupName: string,
+    optionsOrCallback?: CallOptions | RestoreDatabaseCallback,
+    cb?: RestoreDatabaseCallback
+  ): Promise<RestoreDatabaseResponse> | void {
+    const callback =
+      typeof optionsOrCallback === 'function'
+        ? (optionsOrCallback as RestoreDatabaseCallback)
+        : cb;
+    const gaxOpts =
+      typeof optionsOrCallback === 'object'
+        ? (optionsOrCallback as CallOptions)
+        : {};
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IRestoreDatabaseRequest = extend(
+      {
+        parent: this.instance.formattedName_,
+        databaseId: this.id,
+        backup: Backup.formatName_(this.instance.formattedName_, backupName),
+      }
+    );
+    return this.request(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'restoreDatabase',
+        reqOpts,
+        gaxOpts,
+      },
+      (err, resp) => {
+        if (err) {
+          callback!(err, null, resp!, resp!);
+          return;
+        }
+        callback!(null, this, resp, resp);
+      }
```

Comment:
Same here. This should be `(err, operation, resp)` and I tried to make it so but the compiler keeps telling me that `operation` is undefined. I can't work out how to fix this.